packages/runner/lib/server.ts

```diff
@@ -58,14 +67,51 @@ function runProcedure() {
                     input: codeParams
                 });
                 usage.track(nangoProps);
-                return await exec(nangoProps, input.isInvokedImmediately, input.isWebhook, code, codeParams);
+                const scriptType: 'sync' | 'action' | 'webhook' = input.isWebhook ? 'webhook' : input.isInvokedImmediately ? 'action' : 'sync';
+                return await exec(nangoProps, scriptType, code, codeParams);
             } finally {
                 usage.untrack(nangoProps);
                 logger.info('Task done');
             }
         });
 }
 
+function startProcedure() {
+    return publicProcedure
+        .input((input) => input as StartParams) //TODO: zod
+        .mutation(({ input }): boolean => {
+            const { taskId, nangoProps, scriptType, code, codeParams } = input;
+            logger.info('Received task', {
+                taskId: taskId,
+                env: nangoProps.environmentId,
+                connectionId: nangoProps.connectionId,
+                syncId: nangoProps.syncId,
+                input: codeParams
+            });
+            usage.track(nangoProps);
+            // executing in the background and returning immediately
+            // sending the result to the jobs service when done
+            setImmediate(async () => {
+                try {
+                    const { error, response } = await exec(nangoProps, scriptType, code, codeParams);
+                    await httpFetch({
+                        method: 'PUT',
+                        url: `${jobsServiceUrl}/task/${taskId}`,
+                        data: superjson.stringify({
+                            scriptType,
+                            error,
+                            output: response
```

Comment:
I am not 100% of the response schema yet. I might modify it when I implement the PUT `/task/ID` endpoint on `jobs` side